subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/CleanupOutputsStepTest.groovy

```diff
@@ -31,14 +29,11 @@ import org.gradle.internal.fingerprint.FileCollectionFingerprint
 import org.gradle.internal.fingerprint.overlap.OverlappingOutputs
 import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
 import org.junit.Rule
-import spock.lang.Specification
 
-class CleanupOutputsStepTest extends Specification implements FingerprinterFixture {
+class CleanupOutputsStepTest extends StepSpec implements FingerprinterFixture {
     @Rule
     TestNameTestDirectoryProvider temporaryFolder = new TestNameTestDirectoryProvider()
-    def delegate = Mock(Step)
-    def context = Mock(InputChangesContext)
-    def work = Mock(UnitOfWork)
+    final InputChangesContext context = Stub()
```

Comment:
I agree with your reasoning, and it hurts my eye as well.

The reason for the change is that if I keep the original format it complains about overriding `getContext()` from `StepSpec`. The `final` is there to avoid generating a setter, though that's not really crucial. I just wanted to add some keyword prefix to make it look more like the other field declarations, though I admit it's not entirely successful.

Reply:
How about only declaring an abstract getter in the base class? How many tests do not override context? It seems like `work` is not overriden.

Reply:
OK, I ended up refactoring this to make it more clear.